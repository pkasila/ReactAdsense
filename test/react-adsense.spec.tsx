import React from 'react';
import renderer from 'react-test-renderer';
import ReactAdsense from '../src';

describe('ReactAdsense', () => {
  it('Create instance', () => {
    const component = renderer.create(
      <ReactAdsense client="sf" slot="1" className="display-flex" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Children', () => {
    const component = renderer.create(
      <ReactAdsense client="sf" slot="1" className="display-flex">
        <p>Loading...</p>
      </ReactAdsense>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Page level ads', () => {
    const component = renderer.create(
      <ReactAdsense
        client="sf"
        slot="1"
        className="display-flex"
        pageLevelAds={true}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Push to window.adsbygoogle', () => {
    // @ts-ignore
    global.window = {
      adsbygoogle: [],
    };
    const component = renderer.create(<ReactAdsense client="sf" slot="1" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
