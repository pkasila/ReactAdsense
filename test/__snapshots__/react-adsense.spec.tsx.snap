// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactAdsense Children 1`] = `
<ins
  className="adsbygoogle display-flex"
  data-ad-client="sf"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="1"
  data-full-width-responsive="false"
  style={
    Object {
      "display": "block",
    }
  }
>
  <p>
    Loading...
  </p>
</ins>
`;

exports[`ReactAdsense Create instance 1`] = `
<ins
  className="adsbygoogle display-flex"
  data-ad-client="sf"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="1"
  data-full-width-responsive="false"
  style={
    Object {
      "display": "block",
    }
  }
/>
`;

exports[`ReactAdsense Page level ads 1`] = `
<ins
  className="adsbygoogle display-flex"
  data-ad-client="sf"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="1"
  data-full-width-responsive="false"
  style={
    Object {
      "display": "block",
    }
  }
/>
`;

exports[`ReactAdsense Push to window.adsbygoogle 1`] = `
<ins
  className="adsbygoogle "
  data-ad-client="sf"
  data-ad-format="auto"
  data-ad-layout=""
  data-ad-layout-key=""
  data-ad-slot="1"
  data-full-width-responsive="false"
  style={
    Object {
      "display": "block",
    }
  }
/>
`;
